src/qc_ncsa.aplf

```diff
@@ -60,6 +60,45 @@
             dagger ⍵         ⍝ else return transposed conjugate
         }
 
+        ⍝ Measures a vector state
+        ⍝
+        ⍝ ⍵: vector state to measure
+        ⍝ ⍺: indices of qubits to measure, if left empty all of the qubits will be measured
+        ⍝
+        ⍝ Returns a pair of items:
+        ⍝ - An array of tuples where the first item of the tuple is the index of the qubit measured and the second item,
+        ⍝    the measured state.
+        ⍝ - The final vector state 
+        measure ← {
+            n_qubits ← (2⍟1↑(⍴⍵))
+            idxs ← ⍳(1↑⍴⍵)
+            ⍺ ← (⍳n_qubits)-1
+            binary_basis ← #.quapl.circuit.tnsidx n_qubits
+            q_levels ← #.quapl.sng.q0 #.quapl.sng.q1
+
+            measure_helper ← {
+                ⍺≡⍬: ⍵
+                a ← (1↑⍺) + 1
+                past_results vs ← ⍵
+                select ← a⌷[2]binary_basis
+
+                ground_idx ← (~select)/idxs
```

Comment:
Not exactly ground/excited states, find new names

Reply:
are basis_0_idx and basis_1_idx better? I can think of others if not